### PR TITLE
Made Commit and Modification objects hashable so that they can be used in dictionaries and sets

### PR DESCRIPTION
**Pull request by HelgeCPH**
Since I did not know how you organize tests, I put two unittests in a new file. Likely they should be somewhere else but I could not figure out where they would fit best.

### CODE REVIEW

**Comment by ishepard** [comment]
Thanks for the PR! 
It seems that we have a different linter, there are many small refactorings around 😄  that's ok!

LGTM!
I just discovered that the CI doesn't run tests because it's in a different branch. I enabled CI only on master, I need to change that in the future 😄 
Can you confirm that all the tests pass?

File: tests/test_hashable.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+#         diff_and_sc,
+#     )
+#     m2 = Modification(
+#         "dspadini/pydriller/myfile.py", None, ModificationType.ADD, diff_and_sc
+#     )
+
+# assert m1.filename == "mynewfile.py"
+# assert m2.filename == "myfile.py"
+# assert m1 != m2
+# assert m3 == m1
```
Comment (ishepard) [inline tests/test_hashable.py:99]: delete this 😄

File: tests/test_hashable.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    c1 = repo.get_commit("e7d13b0511f8a176284ce4f92ed8c6e8d09c77f2")
+    c2 = repo.get_commit(c1.parents[0])
+    c3 = repo.get_commit("a4ece0762e797d2e2dcbd471115108dd6e05ff58")
+
+    commit_dict = {c1: c1.hash, c2: c2.hash, c3: c3.hash}
+
+    assert type(commit_dict) == dict
+    assert commit_dict[c1] == "e7d13b0511f8a176284ce4f92ed8c6e8d09c77f2"
+    assert commit_dict[c2] == c1.parents[0]
+    assert commit_dict[c3] == "a4ece0762e797d2e2dcbd471115108dd6e05ff58"
```
Comment (ishepard) [inline tests/test_hashable.py:31]: can you add assert commit_dict[c1] != commit_dict[c2]?

File: tests/test_hashable.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+    m1 = c1.modifications[0]
+    m2s = c2.modifications
+
+    mod_dict = {m1: c1, m2s[0]: c2, m2s[1]: c2}
+
+    assert type(mod_dict) == dict
+    assert mod_dict[m1].hash == "e7d13b0511f8a176284ce4f92ed8c6e8d09c77f2"
+    assert mod_dict[m2s[0]].hash == c1.parents[0]
+    assert mod_dict[m2s[1]].hash == c1.parents[0]
```
Comment (ishepard) [inline tests/test_hashable.py:52]: can you add m1 != m2[0]?

**Review by ishepard — CHANGES_REQUESTED** [review]
LGTM! Just some nit comments. Thanks for the good job!

File: tests/test_hashable.py (hunk 1 above)
Comment (HelgeCPH) [reply tests/test_hashable.py]: Arrghh... yes I forgot that. These were the tests that I used for inspiration from your test suite :) Fixed now

File: tests/test_hashable.py (hunk 3 above)
Comment (HelgeCPH) [reply tests/test_hashable.py:53]: Yes, I added cases for this and the comment below

**Comment by HelgeCPH** [comment]
I incorporated your feedback. Sorry for the code formatting, of course I forgot this. My editor applies black on save.
Which code formatter shall I use instead, then I will bring back the code into your format

**Review by ishepard — APPROVED** [review]

**Comment by ishepard** [comment]
Perfect! Thanks for the PR!

**Pull request merged**